File: ai-form-assistant-fastapi/app/modules/users/service.py
from sqlalchemy.orm import Session
import uuid

from .repository import user_repository
from .model import User


class UserService:

    def get_users(self, db: Session):
        return user_repository.get_all(db)

    def get_user(self, db: Session, user_id: uuid.UUID):
        return user_repository.get_by_id(db, user_id)

    def create_user(self, db: Session, user: User) -> User:
        try:
            db_user = user_repository.create(db, user)
            db.commit()
            return db_user
        except Exception:
            db.rollback()
            raise

    def delete_user(self, db: Session, user: User) -> None:
        try:
            user_repository.delete(db, user)
            db.commit()
        except Exception:
            db.rollback()
            raise


user_service = UserService()

File: ai-form-assistant-node/src/modules/users/service.js
const userRepository = require("./repository");
const sequelize = require("../../database/sequelize");

class UserService {

    async getUsers() {
        return userRepository.getAll();
    }

    async getUser(id) {
        return userRepository.getById(id);
    }

    async createUser(data) {
        const t = await sequelize.transaction();
        try {
            const user = await userRepository.create(data, { transaction: t });
            await t.commit();
            return user;
        } catch (error) {
            await t.rollback();
            throw error;
        }
    }

    async deleteUser(id) {
        const t = await sequelize.transaction();
        try {
            const result = await userRepository.delete(id, { transaction: t });
            await t.commit();
            return result;
        } catch (error) {
            await t.rollback();
            throw error;
        }
    }

}

module.exports = new UserService();